### ISSUE TITLE
Incorrect formatting for "Trigger parameterized builds" plugin

### ISSUE THREAD

**Issue by gdhbashton** [voxpupuli/puppet-jenkins_job_builder#11]
Hullo,

I'm using 1.1.0 of your Puppet module (thank you :) and when I provide Puppet hiera YAML for a Jenkins job `genersys`:

```
          publishers:
            - trigger-parameterized-builds:
              - project: "genersys-outgoing-promotion-filewriter. genersys-outgoing-promotion-ftp-reader, genersys-outgoing-promotion-ftp-writer, genersys-outgoing-promotion-service"
                condition: UNSTABLE_OR_BETTER
                predefined-parameters: PKG_VERSION=$PKG_VERSION
```

I get this output in `/tmp/jenkins-genersys.yaml`:

```
       publishers:
        - trigger-parameterized-builds:
           - condition: "UNSTABLE_OR_BETTER"
           - predefined-parameters: "PKG_VERSION=$PKG_VERSION"
           - project: "genersys-outgoing-promotion-filewriter. genersys-outgoing-promotion-ftp-reader, genersys-outgoing-promotion-ftp-writer, genersys-outgoing-promotion-service"

```

The `jenkins-jobs` run fails with `KeyError: 'project'` :(

I found that I need to manipulate the `/tmp/jenkins-genersys.yaml` to look like this in order for it to be processed successfully by `jenkins-jobs`

```
       publishers:
        - trigger-parameterized-builds:
           - project: "genersys-outgoing-promotion-filewriter, genersys-outgoing-promotion-ftp-reader, genersys-outgoing-promotion-ftp-writer, genersys-outgoing-promotion-service"
             predefined-parameters: PKG_VERSION=$PKG_VERSION
             condition: "UNSTABLE_OR_BETTER"
```

Is there a different way in Puppet hiera that I need to express the configuration to get the desired output?

Cheers,
Gavin.

**Comment by madAndroid** [comment]
AIUI, the symptoms you are experiencing are similar to those in #18 .. and this should be addressed with the changes in #19 - with those changes, the structure you define in Hiera is persisted when written to disk, instead of being incorrectly parsed.

**Comment by madAndroid** [comment]
This is fixed with b52901a5a3fc844a65abc72f76fc5bacd45b3f11, which implements a `to_yaml` approach in the template.
